src/vaults/SyncRequestManager.sol

```diff
@@ -177,6 +181,9 @@ contract SyncRequests is BaseInvestmentManager, ISyncRequests {
         auth
         returns (uint256 shares)
     {
+        VaultDetails memory vaultDetails = poolManager.vaultDetails(vault_);
+        require(poolManager.isLinked(vault_.poolId(), vault_.scId(), vaultDetails.asset, vault_), AssetNotAllowed());
+
```

Comment:
Probably should have split up PR due to rename noise. Here is the missing linked vault check for which I also added tests for both managers